nixos/modules/services/misc/recyclarr.nix

```
@@ -0,0 +1,166 @@
+
+    package = lib.mkPackageOption pkgs "recyclarr" { };
+
+    configuration = lib.mkOption {
+      type = format.type;
+      default = { };
+      example = lib.literalExpression ''
+        {
+          sonarr = [{
+            instance_name = "main";
+            base_url = "http://localhost:8989";
+            api_key = {
+              _secret = "/run/credentials/recyclarr.service/sonarr-api_key";
+            };
+          }];
+          radarr = [{
+            instance_name = "main";
+            base_url = "http://localhost:7878";
+            api_key = {
+              _secret = "/run/credentials/recyclarr.service/radarr-api_key";
+            };
+          }];
+        }
+      '';
+      description = lib.mdDoc ''
+        Recyclarr YAML configuration as a Nix attribute set.
+
+        For detailed configuration options and examples, see the
+        [official configuration reference](https://recyclarr.dev/wiki/yaml/config-reference/).
+
+        The configuration is processed using [utils.genJqSecretsReplacementSnippet](https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/utils.nix#L232-L331) to handle secret substitution.
+
+        To avoid permission issues, secrets should be provided via systemd's credential mechanism:
+
+        ```nix
+        systemd.services.recyclarr.serviceConfig.LoadCredential = [
+          "radarr-api_key:''${config.sops.secrets.radarr-api_key.path}"
+        ];
+      '';
+    };
+
+    workingDir = lib.mkOption {
+      type = lib.types.path;
+      default = "/var/lib/recyclarr";
+      description = "Working directory for recyclarr.";
+    };
+
+    schedule = lib.mkOption {
+      type = lib.types.str;
+      default = "daily";
+      description = "When to run recyclarr in systemd calendar format.";
+    };
+
+    command = lib.mkOption {
+      type = lib.types.str;
+      default = "sync";
+      description = "The recyclarr command to run (e.g., sync).";
+    };
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = "recyclarr";
+      description = "User account under which recyclarr runs.";
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      default = "recyclarr";
+      description = "Group under which recyclarr runs.";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    users.users = lib.mkIf (cfg.user == "recyclarr") {
+      recyclarr = {
+        isSystemUser = true;
+        description = "recyclarr user";
+        home = cfg.workingDir;
+        group = cfg.group;
+      };
+    };
+
+    users.groups = lib.mkIf (cfg.group == "recyclarr") {
+      ${cfg.group} = { };
+    };
+
+    systemd.tmpfiles.settings."recyclarr" = {
+      "${cfg.workingDir}".d = {
+        mode = "0750";
+        user = cfg.user;
+        group = cfg.group;
+      };
+    };
+
+    systemd.services.recyclarr = {
+      description = "Recyclarr Service";
+
+      preStart = utils.genJqSecretsReplacementSnippet cfg.configuration configPath;
```

An explicit note about using JSON instead of YAML makes it easier to know it isn't a mistake when reading it later.

```suggestion
      # YAML is a JSON super-set
      preStart = utils.genJqSecretsReplacementSnippet cfg.configuration configPath;
```